Development

madhavigowrisetty opened the pull request:
@caw13

wYaobiz committed:
Add basic class

Add basic class

madhavigowrisetty committed:
removed an attribute

madhavigowrisetty committed:
added java doc on the code

madhavigowrisetty committed:
added java doc on the code

madhavigowrisetty committed:
added java doc on the code

madhavigowrisetty committed:
modified the author name

madhavigowrisetty committed:
modified the author name

madhavigowrisetty committed:
removed information about package

madhavigowrisetty committed:
Java doc added to the code

madhavigowrisetty committed:
Java doc added to the code

madhavigowrisetty committed:
Java doc added to the code

madhavigowrisetty committed:
the java doc is modified

madhavigowrisetty committed:
WarningState.java is removed

madhavigowrisetty committed:
modified doc for setRasberryPiState

Commit:
Update RasberriPiState.java

anilvivek7 committed:
Clas updated

updated the file by adding javadoc

anilvivek7 committed:
updated the file by adding javadoc

anilvivek7 committed:
Added Javadoc

anilvivek7 committed:
Updated the file with javadoc

anilvivek7 committed:
Javadoc added

anilvivek7 committed:
Added javadoc

anilvivek7 committed:
Updated the code and javadoc

anilvivek7 committed:
Updated the code and javadoc

anilvivek7 committed:
Updated the code and javadoc

hiraldepani committed:
Edited Class name

madhavigowrisetty committed:
modified class name

madhavigowrisetty committed:
Update WarningStateLED.java

madhavigowrisetty committed:
Update WarningStateBuzzer.java

madhavigowrisetty committed:
Update OnState.java

madhavigowrisetty committed:
Update OffState.java

hiraldepani committed:
added javadoc

caw13 commented on src/edu/cs505/g5/ObserverBuzzer.java:
```diff
@@ -0,0 +1,21 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/ObserverDHTSensor.java:
```diff
@@ -0,0 +1,15 @@
+package MagicMirror2;
```
You have the package as MagicMirror2, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/ObserverLED.java:
```diff
@@ -0,0 +1,20 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/ObserverNotification.java:
```diff
@@ -0,0 +1,21 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/OffState.java:
```diff
@@ -0,0 +1,16 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/OnState.java:
```diff
@@ -0,0 +1,8 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/RasberriPiState.java:
```diff
@@ -0,0 +1,23 @@
+package MagicMirror2;
```
You have the package as MagicMirror2, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/RasberryPi.java:
```diff
@@ -0,0 +1,25 @@
+
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/RasberryPiStateChange.java:
```diff
@@ -0,0 +1,16 @@
+
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectConcreteTemprature.java:
```diff
@@ -0,0 +1,22 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+p
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/WarningStateBuzzer.java:
```diff
@@ -0,0 +1,19 @@
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/WarningStateLED.java:
```diff
@@ -0,0 +1,20 @@
+
+package edu.ccsu.cs505.g5;
```
You have the package as edu.ccsu.cs505.g5, but your code is in the directory edu.cs505.g5 thus compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/RasberriPiState.java:
```diff
@@ -0,0 +1,23 @@
+ * This class shows different states of RasberriPi
+ * @author CS505-Group5 
+ */
+public class RasberryPiState {
```
Compile error.  Class does not match file name RasberriPiState.java

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+p
```
This character causes a compile error

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+ *
+ * @author CS505-Group5
+ */
+import java.util.Vector;
```
Import is unused so shouldn't be here

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * This class is the Subject class in the observer pattern which will notify the observers if the temperature goes beyond the limit-set
+ */
+public class SubjectDHTSensor {
```
you should have an abstract base class for your sensor which just hold the observers, adding/removing them and the notification method

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Because the SubjectDHTSensor calls the interface Observer and all other classes that implement that interface as well
+     */
+        private ObserverLED ledObserver = new ObserverLED();
```
The Subject base class should NOT have any references to the concrete observers.  It should only have a list of ObserverDHTSensor interfaces.  Based on this design it would force anybody that wanted to use your component to have all of these observers.  You instead want to have none, and allow them to add what they want.

wYaobiz replied:
modified

caw13 commented on src/edu/cs505/g5/SubjectDHTSensor.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * @param o of type ObserverDHTSensor is send as a parameter to add as concrete observer 
+ */
+    public void attch (ObserverDHTSensor o) {
```
Go ahead and spell out the name of "attach" here there is no reason to have an abbreviation.

wYaobiz replied:
changed to addObserver

caw13 commented on src/edu/cs505/g5/ObserverBuzzer.java:
```diff
@@ -0,0 +1,21 @@
+ */
+public class ObserverBuzzer implements ObserverDHTSensor {
+
+	public int observerState;
```
In general don't make any public attributes.  Make the actual attributes private or protected and add accessors (i.e. getters/setters)

caw13 replied:
Note you want to have your concrete Observers have a reference to the concrete class they are interested in.  The notification method just lets it know that what it was listening to changed, then it uses its reference to the concrete observer to look at the details of what changed.

wYaobiz replied:
modified.

caw13 commented on src/edu/cs505/g5/OffState.java:
```diff
@@ -0,0 +1,16 @@
+    /**
+     * This observer sends a notification (to the user) that RaspberryPi is going to switch off
+     */
+    public Vector  myObserverNotification;
```
You should not have any attribute types particularly on a public interface that are of a concrete type.  Use the Collections framework as was discussed in class and unless you explicitly need the Vector class because of multi-threading issues this isn't the right concrete class anyway.

caw13 reviewed (changes requested):
Think about renaming your "g5" package to something that is descriptive of your component.  Also the "getState" "setState" does not necessarily imply state pattern it is just referring to calling a method to modify/read the instance state of the concrete observer.  

There are major compile issues that need to be fixed before this code is merged into the master.

caw13 commented:
@CCSU-CS417F17/group-5 
Also you should update the ReadMe.md to describe to other users/teams how to use your component

wYaobiz committed:
Changed directory

wYaobiz committed:
Changed component basic class, move state to another folder

wYaobiz committed:
folder changed

wYaobiz committed:
Changed to right folder.

madhavigowrisetty committed:
Merge branch 'Development' of https://github.com/CCSU-CS417F17/IAmNotAMirror-public into Development

wYaobiz committed:
implement some functions, except communication with sensor.

madhavigowrisetty committed:
Merge branch 'Development' of https://github.com/CCSU-CS417F17/IAmNotAMirror-public into Development

wYaobiz committed:
Add Document Branch

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Updated javadoc

madhavigowrisetty committed:
Merge branch 'Development' of https://github.com/CCSU-CS417F17/IAmNotAMirror-public into Development

madhavigowrisetty committed:
test case body is added

madhavigowrisetty committed:
added import files

madhavigowrisetty committed:
created objects

madhavigowrisetty committed:
testing the ist observer

madhavigowrisetty committed:
added 2nd observer

madhavigowrisetty committed:
testing the 2nd observer add method

madhavigowrisetty committed:
adding the 2 observers

madhavigowrisetty committed:
removed second observer

madhavigowrisetty committed:
testing the second observer removed or not

madhavigowrisetty committed:
tested the 1st observer presence

madhavigowrisetty committed:
added notify method

Commit:
Adding files for LED Observer Warner Test

Commit:
Added more test methods in LEDObserverWarnerTest

hiraldepani committed:
Added Junit Testing

Added Junit testing for SubjectDHTSensor

hiraldepani committed:
Updated Test case

hiraldepani committed:
Changed Name

hiraldepani committed:
Added more test cases

Added Test construction test case

hiraldepani committed:
Updated Test Case

madhavigowrisetty committed:
modified SubjectSensor

madhavigowrisetty committed:
adding java doc

madhavigowrisetty committed:
added author information in java doc

madhavigowrisetty committed:
added List information in java doc

madhavigowrisetty committed:
added addObserver method information in java doc

madhavigowrisetty committed:
added removeObserver method information in java doc

madhavigowrisetty committed:
added notifyObserver method information in java doc

Commit:
Added warn method

anilvivek7 committed:
Modified code and updated javadoc

madhavigowrisetty committed:
modified readme file

Commit:
Updated with description

madhavigowrisetty committed:
Merge branch 'Development' of https://github.com/CCSU-CS417F17/IAmNotAMirror-public into Development

hiraldepani committed:
Updated Code

Commit:
Updated with interval description

hiraldepani committed:
Added JavaDoc

anilvivek7 committed:
Added few points to the description of reusable component

anilvivek7 committed:
Modified the code and updated the javadoc

hiraldepani committed:
Added description

wYaobiz committed:
update readme

update readme

wYaobiz committed:
Merge branch 'Test' into Development

madhavigowrisetty committed:
added uml class diagram

wYaobiz committed:
After merged from testing

After merged from testing

wYaobiz committed:
Deleted unused files

Deleted unused files

wYaobiz committed:
Generate Javadoc

Generate Javadoc

wYaobiz committed:
Used template Pattern

1. Made the unpdate method as concrete template method. 2. Made Hook Method. 3. Modified LEDObserverWarner internal logic.

madhavigowrisetty committed:
modified the template class to implement template pattern

hiraldepani committed:
Edited Javadoc

hiraldepani committed:
Edited Javadoc

anilvivek7 committed:
Updated Javadoc

hiraldepani committed:
Edited Javadoc

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Javadoc Updated

hiraldepani committed:
Updated Javadoc

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Updated javadoc

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Updated Javadoc

anilvivek7 committed:
javadoc updated

anilvivek7 committed:
Updated Javadoc

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Updated all the files

anilvivek7 committed:
javadoc Updated

anilvivek7 committed:
Updated Javadoc

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Updated Javadoc

anilvivek7 committed:
Javadoc Updated

hiraldepani committed:
Updated explanation about reusable component

anilvivek7 committed:
updated the description

madhavigowrisetty committed:
corrected minor grammer chnages

wYaobiz committed:
Public component sprint 2

1. Removed sensor obj as the parameter of update method.
2. Changed the version number.

hiraldepani committed:
Updated Class Diagram

hiraldepani committed:
Updated Class Diagram

hiraldepani committed:
Updated Class Diagram

hiraldepani committed:
Updated Javadoc

hiraldepani committed:
Updated Javadoc

hiraldepani committed:
Updated Javadoc

hiraldepani committed:
Updated Javadoc

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
Files updated

anilvivek7 committed:
Javadoc Updated

anilvivek7 committed:
javadoc Updated

Commit:
small correction
